crates/viewer/re_space_view_dataframe/src/display_record_batch.rs

```diff
@@ -0,0 +1,332 @@
+        entity_path: &EntityPath,
+        component_name: ComponentName,
+        index: usize,
+    ) {
+        let data = match self {
+            Self::Null => {
+                ui.label("null");
+                return;
+            }
+            Self::ListArray(list_array) => {
+                list_array.is_valid(index).then(|| list_array.value(index))
+            }
+            Self::DictionaryArray { dict, values } => dict
+                .is_valid(index)
+                .then(|| values.value(dict.key_value(index))),
+        };
+
+        if let Some(data) = data {
+            ctx.component_ui_registry.ui_raw(
+                ctx,
+                ui,
+                UiLayout::List,
+                latest_at_query,
+                ctx.recording(),
+                entity_path,
+                component_name,
+                Some(row_id),
+                &*data,
+            );
+        } else {
+            ui.label("-");
+        }
+    }
+}
+
+pub(crate) enum DisplayColumn {
+    RowId {
+        row_id_times: ArrowPrimitiveArray<u64>,
+        row_id_counters: ArrowPrimitiveArray<u64>,
+    },
+    Timeline {
+        timeline: Timeline,
+        time_data: ArrowPrimitiveArray<i64>,
+    },
+    Component {
+        entity_path: EntityPath,
+        component_name: ComponentName,
+        component_data: ComponentData,
+    },
+}
+
+impl DisplayColumn {
+    #[allow(clippy::borrowed_box)] // https://github.com/rust-lang/rust-clippy/issues/11940
+    fn try_new(
+        column_schema: &ColumnDescriptor,
+        column_data: &Box<dyn ArrowArray>,
+    ) -> Result<Self, DisplayRecordBatchError> {
+        match column_schema {
+            ColumnDescriptor::Control(desc) => {
+                if desc.component_name == ComponentName::from("rerun.controls.RowId") {
+                    let row_ids = column_data
+                        .as_any()
+                        .downcast_ref::<ArrowStructArray>()
+                        .expect("sanity checked");
+                    let [times, counters] = row_ids.values() else {
+                        panic!("RowIds are corrupt -- this should be impossible (sanity checked)");
+                    };
+
+                    #[allow(clippy::unwrap_used)]
+                    let row_id_times = times
+                        .as_any()
+                        .downcast_ref::<ArrowPrimitiveArray<u64>>()
+                        .expect("sanity checked")
+                        .clone();
+
+                    #[allow(clippy::unwrap_used)]
+                    let row_id_counters = counters
+                        .as_any()
+                        .downcast_ref::<ArrowPrimitiveArray<u64>>()
+                        .expect("sanity checked")
+                        .clone();
+
+                    Ok(Self::RowId {
+                        //descriptor: desc,
+                        row_id_times,
+                        row_id_counters,
```

Comment:
minor nit: but "times" and "counters" only makes sense for `RowId` as `TUID`. If we switch to UUID these names becomes nonsense. We also don't treat them as times and counters anywhere in this file, so perhaps it would be better to name time `row_id_hi` and `row_id_lo` or something? 🤷

Reply:
Also, why are we storing `RowId` as two arrays of `u64` instead of a single array of `[u8; 16]` ? 🤔

Reply:
I'm basically following what is being done in `re_datastore`. I expect this to be carved in stone with the sorbet spec, so let's adjust the implementation when we have that (fundamentally, the purpose of `DisplayRecordBatch` and friends is to decode a sorbet-compliant `RecordBatch` for the purpose of ui).

Reply:
Related:
- #6818